### DISCUSSION TITLE
head (app/head) doesnt work properly when sharing the page

### DISCUSSION THREAD

**Discussion by handrei** [vercel/next.js#15112] [Help] [answered]
I've migrated from create-react-app to create-next-app to my website for the SEO, but the Head(app/head) component isn't working properly

Assuming I have a blog, and in pages, I have the following structure: pages/blog/index.js, pages/blog/[slug].js. In [slug].js, I'm using the Head component so it takes the title of the blog post: <Head><title>{post.title}</title></Head> The problem is that when using react-share on the blog post it doesn't put the right title, even when I do an inspect element on the page, I can see the right Head values, but the Facebook debug (https://developers.facebook.com/tools/debug/) tool doesn't recognize the right title.
i have to mention that when I'm in the pages/blog/index.js the sharing is working corectly

am i doing something wrong ?

**Comment by illuminist** [comment, marked as answer]
You need to make sure that SSR is going right: check for rendered HTML source(not via element inspector), check if you have fetched data initial props via `getIntialProps`, `getServerSideProps`, or `getStaticProps` correctly and the component is actually using those props.

**Reply by handrei** [reply]
you mean that i have to fetch the data via `getInitialProps` ? i also tried that but with same results. 
right now i have a function getData() which I'm calling it in useEffect hook, but i tried to fetch the data in `getInitialProps` and pass the data through props in component, but no success.

**Reply by handrei** [reply]
```
Post.getInitialProps = async ({ req, query }) => {
  const { origin } = absoluteUrl(req);
  const namespacesRequired = ['common'];

  const { article } = await api.getOne(
    `/article/${query.slug}`
  );

  return { origin, namespacesRequired, article };
};

```
```
const Post = ({ t, origin, article }) => {....
const [post, setPost] = useState(null);
useEffect(() => {
    if (!post) {
      setPost(article);
    }
  }, [post]);
<Head>
          <title>{'News | ' + post.title}</title>
          <meta name='description' content={post.title} key='description' />
          <meta property='og:title' content={post.title} key='title' />
          <meta property='og:image' content={post.cover} key='image' />
          <meta property='og:image:width' content='400' />
          <meta property='og:image:height' content='300' />
          <meta property='og:image:type' content='image/jpeg' />
        </Head>
```

and what i got on the view source(not element inspector) is another title, not from the serverside prop

**Reply by handrei** [reply]
acutally i resolved it, not using useEffect at all, just the article return in the getInitialProps. thanls a lot

**Reply by illuminist** [reply]
That's right.

Many problem can be solve without using useEffect, and that's the ideal solution.
